gradio/themes/default.py

```diff
@@ -0,0 +1,236 @@
+from .utils import Theme
+
+
+class Default(Theme):
+    def __init__(
+        self,
+        *,
+        primary_hue="orange",
+        secondary_hue="blue",
+        neutral_hue="grey",
+    ):
+        # Core values
+        self.color_accent_base = self._color(primary_hue, 500)
+        self.color_accent_soft = self._color(primary_hue, 300)
+        self.color_accent_soft_dark = self._color(neutral_hue, 600)
```

Comment:
Should we have one theme define both light and dark variants?

I think it would be better if we had separate `DefaultLight` and `DefaultDark` and have a custom theme take precedence over the /?__theme query parameter.

I think it would make it easier to create new themes this way and also make sure that the theme I define locally is the theme that gets displayed on Spaces.

For example, right now I have a local theme that looks good on dark mode:

<img width="1262" alt="image" src="https://user-images.githubusercontent.com/41651716/215088969-9b8fa3cb-8d65-4cfb-9241-d97e9f19b547.png">

But now I have to spend 2x as much time trying to get this to work in light mode, especially since spaces forces "light" theme, when I just want the dark version to be displayed everywhere.

WDTY? cc: @pngwn

Reply:
hmm I would think dark and light modes are variants of a single theme, but I can see it either way. I'll let @pngwn decide

Reply:
Wouldn't it make more sense for a "theme" should consist of both a dark mode and a light mode, since a user's settings on Spaces will determine whether the dark mode or light mode gets shown. @freddyaboulton what if a user has light mode enabled and opens up a demo with your theme on it?

Reply:
Ah I actually didn’t know you could set light vs dark preference on spaces. I thought you always got light theme lol.

My concern is that some theme creators may only want to make a theme based on darker or (lighter) colors. In my example above, I “branched” off the default theme and only tweaked some of the attributes that end in _dark. That’s how I would like my app to be styled regardless of where it’s hosted. 

I don’t want to have to think about a light version of my theme (cause I want it to use dark colors anyway) or manually apply the same tweaks to the “light” Theme attributes (tedious).

If we feel strongly that “light” and “dark” are variants of the same theme, we can add an api to make it easier to apply the same theme choices to both light and dark variants?

Reply:
It's not just a matter of settings on Spaces. Gradio inherits from your browser settings so if you have light mode enabled in your browser, you'll see the light theme and vice versa. For many users, light/dark mode is a matter of accessibility so I strongly think we should require both dark mode and light mode to be supported for every theme. My understanding right now is that if you don't override the default light theme parameters, you'll end up with the default light theme, which is a reasonable behavior IMO if you don't implement a light theme yourself

Reply:
I am convinced. Light and dark or (base and dark) should be the same theme.  Thank you for talking this through with me @abidlabs @aliabid94 @pngwn !